xbmc/addons/AddonInstaller.cpp

```diff
@@ -565,20 +575,20 @@ bool CAddonInstallJob::DoWork()
 
       // at this point we have the package - check that it is valid
       SetText(g_localizeStrings.Get(24077));
-      if (!m_hash.empty())
+      if (!hash.empty())
       {
-        md5 = CUtil::GetFileDigest(package, KODI::UTILITY::CDigest::Type::MD5);
-        if (!StringUtils::EqualsNoCase(md5, m_hash))
+        std::string actualHash = CUtil::GetFileDigest(package, hashType);
```

Comment:
`hashType` may be used uninitialized

Reply:
either both `hash` and `hashType` are set or neither is

Reply:
OK so the compiler doesn't understand this... I'll see what I can do.